task2/js/script.js

```diff
@@ -0,0 +1,130 @@
+function isArray(something) {
+  return something instanceof Array;
+}
+
+function isArray2(something) {
+  return Object.prototype.toString.call(something) === '[object Array]';
+}
+
+function range(from, to, step) {
+  var result = [];
+
+  if (from == undefined && to == undefined && step == undefined)
+    return result;
+
+  step = step || 1;
+
+  if (step < 0 && from < step)
+    return result;
+
+  if (to == undefined && from != undefined) {
+    to = from;
+    from = 0;
+  }
+  else if (from == undefined && to != undefined) {
+    from = 0;
+  }
+
+  var countElements = parseInt((to - from) / step);
+
+  for (var i = 0; i <= countElements; i++)
+    result.push(from + i * step);
+
+  return result;
+}
+
+function compact(source) {
+  return source.filter(function (value) {
+    return value;
+  });
+}
+
+function compact2(source) {
+  var result = [];
+
+  for (var i = 0; i < source.length; i++)
+    if (source[i])
+      result.push(source[i]);
+
+  return result;
+}
+
+function sum(source) {
+
+  return source.reduce(function (sum, current) {
+    return sum + current;
+  }, 0);
+}
+
+function sum2(source) {
+  var result = 0;
+
+  for (var i = 0; i < source.length; i++)
+    result += source[i];
+
+  return result;
+}
+
+function unique(source) {
+  var obj = {};
+  var result = [];
+
+  for (var i = 0; i < source.length; i++)
+    source[i] == '[object Object]' ?
+        obj[JSON.stringify(source[i])] = source[i] :
+        obj[source[i]] = source[i];
+
+  for (i in obj) {
+    result.push(obj[i])
+  }
+
+  return result;
+}
+
+function last(source) {
+  return source[source.length - 1];
+}
+
+function excludeLast(source, count) {
+
+  count = count || 1;
```

Comment:
if count is 0 it will be 1
it depends of you want, I mean support 0 or not
source.slice(0, source.length - -5) = source + [undef,undef,undef,undef,undef]